src/player.cpp

```diff
@@ -3751,6 +3751,18 @@ void Player::addAttacked(const Player* attacked)
 	attackedSet.insert(attacked->guid);
 }
 
+void Player::removeAttacked(const Player* attacked)
+{
+	if (!attacked || attacked == this) {
+		return;
+	}
+
+	auto it = attackedSet.find(attacked->guid);
```

Comment:
Note that `hasAttacked` that is called before this function already performs this check. You probably don't need to check `hasAttacked` then.